src/cpp/publisher/PublisherImpl.cpp

```diff
@@ -481,3 +498,36 @@ void PublisherImpl::lifespan_expired()
     lifespan_timer_.update_interval_millisec((double)duration_cast<milliseconds>(interval).count());
     lifespan_timer_.restart_timer();
 }
+
+void PublisherImpl::get_liveliness_lost_status(LivelinessLostStatus &status)
+{
+    std::unique_lock<std::recursive_timed_mutex> lock(mp_writer->getMutex());
```

Comment:
Well, I think a mutex is necessary here because the user could be requesting the `LivelinessLostStatus` at the same time as the liveliness timer expires and updates that struct on the writer. Note that in `WLP` I am also locking the `Endpoint` mutex to protect this struct.

In other cases we also seem to be recycling this mutex (for instance, in `HeartbeatResponseDelay`, `InitialAckNack`, ... It was also used when we added the deadline QoS, see for instance `SubscriberImpl::deadline_missed()` and `SubscriberImpl::get_requested_deadline_missed_status()`) so I thought I would do the same here. Should we go for a different approach?

Reply:
Any thoughts?

Reply:
I believe this conversation is complex enough to be discussed in depth, and this isn't the place for it :).